src/main/java/de/florianmichael/viafabricplus/injection/mixin/fixes/minecraft/network/MixinPlayerListEntry.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * This file is part of ViaFabricPlus - https://github.com/FlorianMichael/ViaFabricPlus
+ * Copyright (C) 2021-2024 FlorianMichael/EnZaXD <florian.michael07@gmail.com> and RK_01/RaphiMC
+ * Copyright (C) 2023-2024 contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.florianmichael.viafabricplus.injection.mixin.fixes.minecraft.network;
+
+import net.minecraft.client.network.PlayerListEntry;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+
+import java.util.function.IntSupplier;
+
+@Mixin(PlayerListEntry.class)
+public abstract class MixinPlayerListEntry implements IntSupplier {
```

Comment:
I am not sure if it's a good idea to use IntSupplier, probably nothing other mods would also do but I wanna go the safe way here, can you introduce another interface with viaFabricPlus$getIndex()?